### ISSUE TITLE
ArcGIS JS API 4.9.3 + Angular 7 -- Build fails with "Module not found" Errors

### ISSUE THREAD

**Issue by bruhaha1313** [Esri/arcgis-webpack-plugin#29]
I simply can't get this to work with even a basic map. I've successfully configured @angular-builders/custom-webpack to add the following to the Webpack build:

const ArcGISPlugin = require("@arcgis/webpack-plugin");
module.exports = {
    plugins: [new ArcGISPlugin()]
};

And I'm only importing the minimum components needed to display a basic map:

import EsriMap = require("esri/Map");
import EsriMapView = require("esri/views/MapView");

The build starts off fine and seems to correctly invoke the Dojo Webpack Builder, but dies at the end with multiple low-level "Module not found" errors:

Dojo loader not specified in options.  Building the loader...
processing profile resource E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo-webpack-plugin/buildDojo/loader.profile.js
info(107) Package Version: package: dojo; version: 1.14.0
processing profile resource E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo/dojo.profile.js
info(107) Package Version: package: util; version: 1.14.0
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(209) Missing or empty package.json. filename: E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo-util/build/package.json
processing profile resource E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo-webpack-plugin/buildDojo/loader.profile.js
info(107) Package Version: package: dojo; version: 1.14.0
processing profile resource E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo/dojo.profile.js
info(107) Package Version: package: util; version: 1.14.0
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(209) Missing or empty package.json. filename: E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo-util/build/package.json
discovering resources...
starting reading resources...
starting processing raw resource content...
starting tokenizing resource...
starting processing resource tokens...
starting parsing resource...
starting processing resource AST...
starting executing global optimizations...
starting writing resources...
starting cleaning up...
starting reporting...
Report written to C:/Users/bemarino/AppData/Local/Temp/4/tmp-981652Ig2rQLDbSO/build-report.txt
Process finished normally.
        errors: 0
        warnings: 2
        build time: 0.086 seconds
10% building modules 0/1 modules 1 active ...\CE.MobileETR.Web.ClientApp\src\ma
10% building modules 1/2 modules 1 active ...obileETR.Web.ClientApp\src\polyfil
10% building modules 2/3 modules 1 active ...node_modules\arcgis-js-api\css\mai
snip...
chunk {86} 86.js, 86.js.map () 19 kB  [rendered]
chunk {87} 87.js, 87.js.map () 141 kB  [rendered]
chunk {88} 88.js, 88.js.map () 18.1 kB  [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 15 MB [initial] [rendered]
ERROR in ./node_modules/arcgis-js-api/geometry/support/pe-wasm.js
Module not found: Error: Can't resolve 'fs' in 'E:\p\CE.MobileETR\CE.MobileETR.Web.ClientApp\node_modules\arcgis-js-api\geometry\support'
ERROR in ./node_modules/arcgis-js-api/geometry/support/pe-wasm.js
Module not found: Error: Can't resolve 'path' in 'E:\p\CE.MobileETR\CE.MobileETR.Web.ClientApp\node_modules\arcgis-js-api\geometry\support'
 
I have no idea what to do to fix this. Note that the application worked fine with the older esri-loader package.

**Comment by bruhaha1313** [comment]
I've traced the issue down by running the following simple code in a brand new Angular app (created with "ng new"):

import { Component, OnInit } from '@angular/core';
import EsriMap = require("esri/Map");

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit {
  title = 'esritest';

  ngOnInit() {
    new EsriMap();
  }

}

Trying to build this is enough to reproduce the build failure I identified above.

**Comment by andygup** [comment]
Currently, we only recommend using `esri-loader` for Angular 6+. 

Just a quick peruse thru the error dump above is warning there are some configuration issue(s):

```
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(209) Missing or empty package.json. filename: E:/p/CE.MobileETR/CE.MobileETR.Web.ClientApp/node_modules/dojo-util/build/package.json
```

**Comment by tomwayson** [comment]
https://github.com/Esri/esri-loader/#angular

**Comment by bruhaha1313** [comment]
I know about esri-loader and do have it working successfully, but I was really hoping to use the new WebPack plugin for the benefits over esri-loader that have been discussed elsewhere.

If the warnings indicate something that I'm doing wrong, I'd love to know what. "dojo-util" isn't something I'm using directly, so I would assume it and all its dependencies get loaded when I "npm install".

I did manage to get past my first issue -- I needed to revise my @angular-builders/custom-webpack config file to set fs to 'empty':

const ArcGISPlugin = require('@arcgis/webpack-plugin');
module.exports = {
    plugins: [new ArcGISPlugin({
        locales: ['en'],
        features: {
            "3d": false
        }
    })],

    node: {
        process: false,
        global: false,
        fs: 'empty'
    }
};

The problem now is that if I set "global" to "false", I get "global is not defined" from Kendo's Angular controls, and the application fails to start. But if I leave it defaulted to "true", it's the Esri controls that give me "Module not found: ./../../webpack/buildin/global.js" (and the application fails to start).
